frontend/playground-frontend/src/components/helper_functions/TalkWithBackend.js

```diff
@@ -2,17 +2,20 @@ import { toast } from "react-toastify";
 import { auth } from "../../firebase";
 import axios from "axios";
 
-const uploadToBackend = async (data) => {
+async function uploadToBackend(data) {
   let headers = auth.currentUser
     ? { Authorization: "bearer " + (await auth.currentUser.getIdToken(true)) }
     : undefined;
 
   await axios.post("/api/upload", data, { headers });
-};
+}
 
-const sendToBackend = async (route, data) => {
+async function sendToBackend(route, data) {
   let headers = auth.currentUser
-    ? { Authorization: "bearer " + (await auth.currentUser.getIdToken(true)) }
+    ? {
+        Authorization: "bearer " + (await auth.currentUser.getIdToken(true)),
+        uid: auth.currentUser.uid,
```

Comment:
what makes this function different from previously?

Reply:
Previously the header only contains the auth token but now the header contains both the token and UID